packages/mui-joy/src/Button/Button.test.tsx

```diff
@@ -113,6 +113,18 @@ describe('Joy <Button />', () => {
       const button = getByRole('button');
       expect(button).to.have.property('disabled', true);
     });
+    it('should disable button when loading is true and disabled is false', () => {
+      const { getByRole } = render(<Button loading disabled={false} />);
+
+      const button = getByRole('button');
+      expect(button).to.have.property('disabled', true);
+    });
+    it('should disable button when loading is false and disabled is true', () => {
```

Comment:
This tests passes in master branch too but i added so that it acts as wall when someone accidently tries to remove `inProps.disabled` from below line

https://github.com/mui/material-ui/blob/c26dcb01a973b9827df739508877eca062528339/packages/mui-joy/src/Button/Button.tsx#L219-L220